timer/src/test/java/com/toggl/timer/startedit/domain/WheelChangedEndTimeActionTests.kt

```diff
@@ -0,0 +1,150 @@
+import com.toggl.timer.exceptions.EditableTimeEntryDoesNotHaveAStartTimeSetException
+import io.kotlintest.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.converter.ConvertWith
+import org.junit.jupiter.params.converter.SimpleArgumentConverter
+import org.junit.jupiter.params.provider.ValueSource
+import org.threeten.bp.Duration
+import org.threeten.bp.OffsetDateTime
+import org.threeten.bp.ZoneOffset
+
+@ExperimentalCoroutinesApi
+@DisplayName("The WheelChangedEndTime action")
+class WheelChangedEndTimeActionTests {
+    val initialState = createInitialState()
+    val timeService = mockk<TimeService>()
+    val now = OffsetDateTime.of(2020, 1, 1, 10, 0, 0, 0, ZoneOffset.UTC)
+    val reducer = createReducer(timeService = timeService)
+
+    init {
+        every { timeService.now() } returns now
+    }
+
+    private fun createEditableTimeEntry(startTime: OffsetDateTime?, duration: Duration?) =
+        EditableTimeEntry.empty(1).copy(startTime = startTime, duration = duration)
+
+    @Test
+    fun `throws EditableTimeEntryDoesNotHaveAStartTimeSetException the time entry is new`() = runBlockingTest {
+        val initialTimeEntry = createEditableTimeEntry(null, null)
+        val durationInputted = Duration.ofMinutes(30)
+
+        reducer.testReduceException(
+            initialState.copy(editableTimeEntry = initialTimeEntry),
+            StartEditAction.WheelChangedEndTime(durationInputted),
+            EditableTimeEntryDoesNotHaveAStartTimeSetException::class.java
+        )
+    }
+
+    @Test
+    fun `throws EditableTimeEntryDoesNotHaveADurationSetException the time entry is new when the time entry is running`() = runBlockingTest {
+        val initialTimeEntry = createEditableTimeEntry(now - Duration.ofSeconds(10), null)
+        val durationInputted = Duration.ofMinutes(30)
+
+        reducer.testReduceException(
+            initialState.copy(editableTimeEntry = initialTimeEntry),
+            StartEditAction.WheelChangedEndTime(durationInputted),
+            EditableTimeEntryDoesNotHaveADurationSetException::class.java
+        )
+    }
+
+    @Test
+    fun `sets the duration inputted to the editableTimeEntry's duration when the time entry is stopped`() = runBlockingTest {
+        val initialStartTime = now - Duration.ofHours(1)
+        val initialDuration = Duration.ofMinutes(30)
+        val initialTimeEntry = createEditableTimeEntry(initialStartTime, initialDuration)
+        val durationInputted = Duration.ofHours(1).plusMinutes(30)
+
+        reducer.testReduceState(
+            initialState.copy(editableTimeEntry = initialTimeEntry),
+            action = StartEditAction.WheelChangedEndTime(durationInputted)
+        ) {
+            it shouldBe it.copy(
```

Comment:
I think you meant this?
```suggestion
            it shouldBe initialState.copy(
```

Reply:
YES! I fixed this on the other branch, but I've wrote a bunch of those. Let me fix them all.